src/main/java/io/github/anthorx/parquet/sql/read/SQLParquetReaderWrapper.java

```diff
@@ -17,47 +17,102 @@
 
 import io.github.anthorx.parquet.sql.record.Record;
 import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
 import org.apache.hadoop.fs.Path;
 import org.apache.parquet.hadoop.ParquetFileReader;
 import org.apache.parquet.hadoop.ParquetReader;
 import org.apache.parquet.hadoop.api.ReadSupport;
 import org.apache.parquet.hadoop.util.HadoopInputFile;
+import org.apache.parquet.hadoop.util.HiddenFileFilter;
 import org.apache.parquet.io.InputFile;
 import org.apache.parquet.schema.MessageType;
 import org.apache.parquet.schema.Type;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
 import java.util.List;
 import java.util.stream.Collectors;
 
 public class SQLParquetReaderWrapper {
 
-  private final ParquetReader<Record> parquetReader;
-  private final MessageType schema;
+  private ParquetReader<Record> currentParquetReader;
+  private final Iterator<ParquetReader<Record>> parquetReaderIterator;
+  private MessageType schema;
 
-  public SQLParquetReaderWrapper(String filePath) throws IOException {
-    InputFile inputFile = HadoopInputFile.fromPath(new Path(filePath), new Configuration());
+  public SQLParquetReaderWrapper(String stringFilePath) throws IOException {
+    Path filePath = new Path(stringFilePath);
+    Configuration configuration = new Configuration();
+    FileSystem fileSystem = filePath.getFileSystem(configuration);
+    FileStatus fileStatus = fileSystem.getFileStatus(filePath);
 
+    if(fileStatus.isDir()) {
+      this.parquetReaderIterator = createReadersFromFolderAndInitSchema(fileSystem, filePath, configuration);
+    } else {
+      this.parquetReaderIterator = createReadersFromFileAndInitSchema(fileStatus, configuration);
+    }
+  }
+
+  private Iterator<ParquetReader<Record>> createReadersFromFolderAndInitSchema(FileSystem fileSystem, Path filePath,
+                                                                               Configuration configuration) throws IOException {
+    List<ParquetReader<Record>> parquetReaderList = new ArrayList<>();
+    for (FileStatus currentFileStatus : fileSystem.listStatus(filePath, HiddenFileFilter.INSTANCE)) {
+      InputFile currentInputFile = HadoopInputFile.fromStatus(currentFileStatus, configuration);
+
+      if (schema == null) {
+        this.initFileSchema(currentInputFile);
```

Comment:
Parquet support schema evolution, it means that for a parquet file, partitions can have different schemas (but compatibles).  
So by setting the schema according to the first readed file, we can have side effects. A common approach could be "schema merging", create a single schema based on schema of each partition. It's expensive becatse it required to open/close each partition file.   
We can manage this case in other PR, schema merging is not mandaory to read partitioned files.

Reply:
Ok, was not aware of this. I will create an Issue with that enhancement. Thanks for the tip!